win32ss/user/ntuser/msgqueue.c

```diff
@@ -1235,6 +1235,9 @@ co_MsqSendMessage(PTHREADINFO ptirec,
       WaitObjects[1] = pti->pEventQueueServer;     // Wait 1
       WaitObjects[2] = ptirec->pEThread;           // Wait 2
 
+      /* We need to take a reference, so it doesn't go away! */
+      ObReferenceObject(WaitObjects[2]);
```

Comment:
Just to be sure, the two other objects (especially the pEventQueueServer) can never go away behind our back?

Reply:
`pti` refers to the current thread.
`Message` is freed right after its completionEvent is set, so that interaction is questionable. But it seems unrelated to this issue.

Reply:
The thread can go away. It's also a problem.